client/app/components/dashboards/dashboard-widget/VisualizationWidget.jsx

```diff
@@ -288,6 +304,10 @@ class VisualizationWidget extends React.Component {
     const { localParameters } = this.state;
     const widgetQueryResult = widget.getQueryResult();
     const isRefreshing = isLoading && !!(widgetQueryResult && widgetQueryResult.getStatus());
+    const onParametersEdit = parameters => {
+      widget.options.paramOrder = map(parameters, "name");
+      widget.save();
```

Comment:
```suggestion
      const paramOrder = map(parameters, "name");
      widget.options.paramOrder = paramOrder;
      widget.save("options", { paramOrder });
```

I didn't test, but this should be the syntax to edit only the `options`.

Apart from that, there may be some reason to include this in the `useEditModeHandler` hook, e.g.: debounce the widget save request and reuse the same state we have for Layout change. However, since this is a WIP, we can look into that later (I also didn't test myself yet how this will behave in the UI).